### PR TITLE
fix: minor typos

### PR DESCRIPTION
**Pull request by nschonni**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] New rule
- [ ] Changes an existing rule
- [ ] Add autofixing to a rule
- [x] Other, please explain:

**What changes did you make? (Give an overview)**

Used cSpell to find a few minor typos

### CODE REVIEW

File: __tests__/no-promise-in-callback.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ ruleTester.run('no-promise-in-callback', rule, {
     // arrow functions and other things
     'go(() => Promise.resolve(4))',
     'go((errrr) => a.then(b))',
```
Comment (nschonni) [inline __tests__/no-promise-in-callback.js:22]: Not sure why the testing was unhappy when I tried changing this to `err/error/errr`, but I reverted it
```
    Message:
      Should have no errors but had 1: [
      {
        ruleId: 'no-promise-in-callback',
        severity: 1,
        message: 'Avoid using promises inside of callbacks.',
        line: 1,
        column: 13,
        nodeType: 'MemberExpression',
        endLine: 1,
        endColumn: 19
      }
    ]
```

File: rules/no-native.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ module.exports = {
     /**
      * Checks for and reports reassigned constants
      *
-     * @param {Scope} scope - an escope Scope object
+     * @param {Scope} scope - an scope Scope object
```
Comment (ota-meshi) [inline rules/no-native.js:41]: I think `escope` is an npm package. So I don't think it's a typo.
https://www.npmjs.com/package/escope

```suggestion
     * @param {Scope} scope - an escope Scope object
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: rules/no-native.js (hunk 2 above)
Comment (nschonni) [reply rules/no-native.js:41]: Maybe that was the package some time ago, looking at the blame, that comment was back from ESLint 3 days. It seems like it's an internal definition https://github.com/nschonni/eslint/blob/757cd9d13160f045fbd0a30100eb0913e8229bfa/docs/src/developer-guide/working-with-rules.md#L164 so I'm not sure this comment is very helpful anymore

Comment (ota-meshi) [reply rules/no-native.js:41]: ESLint currently uses the `eslint-scope` package. The `eslint-scope` is a fork of `escope`.
https://github.com/eslint/eslint-scope

If you want to fix the JSDoc for current state, I think the correct way to do it is:
```suggestion
     * @param {Scope} scope - an eslint-scope Scope object
```

**Review by ota-meshi — APPROVED** [review]
Thank you!

**Review by SuzukiSteveTB — CHANGES_REQUESTED** [review]
Cease and desist

**Pull request merged**